pkg/controller/validation/ippool_test.go

```diff
@@ -0,0 +1,644 @@
+// Copyright 2025 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"net/netip"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	crdv1beta1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+)
+
+func TestGetIPRangeSet(t *testing.T) {
+	tests := []struct {
+		name     string
+		ipRanges []crdv1beta1.IPRange
+		expected sets.Set[string]
+	}{
+		{
+			name:     "empty input",
+			ipRanges: []crdv1beta1.IPRange{},
+			expected: sets.New[string](),
+		},
+		{
+			name: "CIDR only",
+			ipRanges: []crdv1beta1.IPRange{
+				{CIDR: "192.168.1.0/24"},
+			},
+			expected: sets.New("192.168.1.0/24"),
+		},
+		{
+			name: "start-end only",
+			ipRanges: []crdv1beta1.IPRange{
+				{Start: "10.0.0.1", End: "10.0.0.5"},
+			},
+			expected: sets.New("10.0.0.1-10.0.0.5"),
+		},
+		{
+			name: "mixed CIDR and start-end",
+			ipRanges: []crdv1beta1.IPRange{
+				{CIDR: "192.168.1.0/24"},
+				{Start: "10.0.0.1", End: "10.0.0.5"},
+				{Start: "2001:db8::1", End: "2001:db8::5"},
+			},
+			expected: sets.New(
+				"192.168.1.0/24",
+				"10.0.0.1-10.0.0.5",
+				"2001:db8::1-2001:db8::5",
+			),
+		},
+		{
+			name: "CIDR takes precedence over start-end",
+			ipRanges: []crdv1beta1.IPRange{
+				{
+					CIDR:  "192.168.1.0/24",
+					Start: "10.0.0.1",
+					End:   "10.0.0.5",
+				},
+			},
+			expected: sets.New("192.168.1.0/24"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetIPRangeSet(tt.ipRanges)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestParseIPRangeCIDR(t *testing.T) {
+	tests := []struct {
+		name         string
+		cidrStr      string
+		expectedCidr string
+		expectedErr  string
+	}{
+		{
+			name:         "valid IPv4 CIDR",
+			cidrStr:      "192.168.1.0/24",
+			expectedCidr: "192.168.1.0/24",
+		},
+		{
+			name:         "valid IPv6 CIDR",
+			cidrStr:      "2001:db8::/64",
+			expectedCidr: "2001:db8::/64",
+		},
+		{
+			name:        "invalid CIDR format",
+			cidrStr:     "192.168.1.1/33",
+			expectedErr: "invalid cidr 192.168.1.1/33",
+		},
+		{
+			name:        "invalid ipv6 cidr",
```

Comment:
nit: you capitalized `CIDR` in other test names

Reply:
done